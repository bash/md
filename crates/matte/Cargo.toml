[package]
name = "matte"
version = "0.1.0"
edition = "2021"

[dependencies]
anstyle.workspace = true
bat.workspace = true
fmtastic.workspace = true
hostname.workspace = true
matte-language-names = { path = "../matte-language-names" }
pulldown-cmark.workspace = true
smallvec.workspace = true
trait-set.workspace = true
unicode-linebreak-chunked.workspace = true
unicode-width.workspace = true
url.workspace = true

[dev-dependencies]
insta = { workspace = true, features = ["glob"] }
serde = { workspace = true, features = ["derive"] }
toml.workspace = true
