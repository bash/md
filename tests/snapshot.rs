use insta::{assert_snapshot, glob};
use matte::{render, supported_parser_options, Options};
use pulldown_cmark::Parser;
use std::fs::read_to_string;

#[test]
fn test_snippets() {
    glob!("snippets/*.md", |path| {
        let markdown = read_to_string(path).unwrap();
        let rendered = render_to_string(&markdown);
        assert_snapshot!(rendered);
    })
}

fn render_to_string(input: &str) -> String {
    let parser = Parser::new_ext(input, supported_parser_options());
    let mut buffer = Vec::new();
    let options = Options::plain_text(120);
    render(parser, &mut buffer, options).unwrap();
    String::from_utf8(buffer).unwrap()
}
