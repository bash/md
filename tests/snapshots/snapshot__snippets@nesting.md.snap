---
source: tests/snapshot.rs
expression: rendered
input_file: tests/snippets/nesting.md
---
[1m[4m[32m[0m[1m[4m[32mNesting[0m

â”ƒ[0m [0mLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore [0m
â”ƒ[0m [0mmagna aliqua. Vitae tempus quam pellentesque nec nam aliquam sem et. Dui nunc mattis enim ut tellus elementum [0m
â”ƒ[0m [0msagittis vitae. Feugiat pretium nibh ipsum consequat nisl vel pretium lectus. Turpis massa sed elementum tempus [0m
â”ƒ[0m [0megestas sed sed risus pretium. Ac turpis egestas sed tempus urna et pharetra. Pellentesque habitant morbi [0m
â”ƒ[0m [0mtristique senectus et netus et malesuada. Volutpat consequat mauris nunc congue nisi vitae suscipit tellus. [0m
â”ƒ[0m [0mPretium aenean pharetra magna ac placerat. Vitae suscipit tellus mauris a.[0m
â”ƒ[0m [0m
â”ƒ[0m [0mâ—ˆâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â—ˆ
â”ƒ[0m [0m
â”ƒ[0m [0mâ”ƒ[0m [0mA condimentum vitae sapien pellentesque. Quis auctor elit sed vulputate mi sit. Iaculis at erat pellentesque [0m
â”ƒ[0m [0mâ”ƒ[0m [0madipiscing commodo elit at imperdiet. Duis tristique sollicitudin nibh sit amet commodo nulla facilisi [0m
â”ƒ[0m [0mâ”ƒ[0m [0mnullam. Mauris a diam maecenas sed enim ut. Et pharetra pharetra massa massa. Facilisi morbi tempus iaculis [0m
â”ƒ[0m [0mâ”ƒ[0m [0murna id volutpat lacus laoreet non. Euismod elementum nisi quis eleifend quam adipiscing. Placerat orci nulla [0m
â”ƒ[0m [0mâ”ƒ[0m [0mpellentesque dignissim enim sit amet venenatis urna. Magna sit amet purus gravida quis blandit turpis cursus [0m
â”ƒ[0m [0mâ”ƒ[0m [0min. Erat nam at lectus urna duis. Sed vulputate mi sit amet. Mattis ullamcorper velit sed ullamcorper morbi. [0m
â”ƒ[0m [0mâ”ƒ[0m [0mPosuere lorem ipsum dolor sit amet consectetur adipiscing elit duis. Est ultricies integer quis auctor elit [0m
â”ƒ[0m [0mâ”ƒ[0m [0msed vulputate. Sed faucibus turpis in eu mi. Mauris a diam maecenas sed enim. Sed turpis tincidunt id aliquet [0m
â”ƒ[0m [0mâ”ƒ[0m [0mrisus feugiat in.[0m
â”ƒ[0m [0mâ”ƒ[0m [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0mEget est lorem ipsum dolor sit amet consectetur adipiscing elit. Varius vel pharetra vel turpis nunc [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0meget. Consequat semper viverra nam libero justo laoreet sit amet cursus. Ac placerat vestibulum lectus [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0mmauris ultrices. Rhoncus urna neque viverra justo nec ultrices. Cursus sit amet dictum sit amet justo [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0mdonec enim diam. Eu consequat ac felis donec et odio pellentesque. Lacus suspendisse faucibus interdum [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0mposuere lorem ipsum. Laoreet sit amet cursus sit amet. Congue eu consequat ac felis donec et odio [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0mpellentesque diam. Arcu non sodales neque sodales. Arcu felis bibendum ut tristique et. Felis bibendum [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0mut tristique et egestas quis ipsum. Arcu vitae elementum curabitur vitae nunc sed velit dignissim [0m
â”ƒ[0m [0mâ”ƒ[0m [0mâ”ƒ[0m [0msodales. Feugiat pretium nibh ipsum consequat nisl. Neque gravida in fermentum et sollicitudin ac.[0m

[1m[32mâ”ƒ[0m [0m[1m[32m1. [0m[1m[32mCode Block inside Quote[0m
â”ƒ[0m [0m
â”ƒ[0m [0m[35mfn[0m [34mmain[0m() {
â”ƒ[0m [0m    println!([32m"[0m[32mhello world[0m[32m"[0m);
â”ƒ[0m [0m}
